### PR TITLE
Add docs for environment setup on XPU device

### PR DESCRIPTION
**Pull request by zhouyu5**
This is a follow-up PR for issues #9384, which aim at providing detailed environment setup guide on Intel GPUs.

### CODE REVIEW

**Comment by chaojun-zhang** [comment]
LGTM

**Comment by zhouyu5** [comment]
@rusty1s @wsad1 Pls have a review.

File: benchmark/multi_gpu/training/README.md [hunk 1]
```diff
@@ -10,15 +10,42 @@ python training_benchmark_cuda.py --dataset ogbn-products --model edge_cnn --num
...
+### Prerequisites
+
+- Intel Data Center GPU Max Series. You could try it through [Intel DevCloud](https://www.intel.com/content/www/us/en/developer/tools/devcloud/services.html).
+- Verify the Intel GPU Driver is installed, refer to the [guide](https://dgpu-docs.intel.com/driver/installation.html).
+
+### docker setup
+
+If you want to run your scripts inside a docker image, you could refer to the [dockerfile](https://github.com/pyg-team/pytorch_geometric/blob/master/docker/Dockerfile.xpu) and the corresponding [guide](https://github.com/pyg-team/pytorch_geometric/blob/master/docker).
+
+### bare-mental setup
```
Comment (DamianSzwichtenberg) [inline benchmark/multi_gpu/training/README.md:24]:
```suggestion
### bare-metal setup
```

File: benchmark/multi_gpu/training/README.md [hunk 2]
```diff
@@ -10,15 +10,42 @@ python training_benchmark_cuda.py --dataset ogbn-products --model edge_cnn --num
...
+
+If you prefer to run your scripts directly on the bare-mental server. We recommend this [guide](https://intel.github.io/intel-extension-for-pytorch/xpu/latest/tutorials/features/DDP.html) and summarize it as follows:
```
Comment (DamianSzwichtenberg) [inline benchmark/multi_gpu/training/README.md:26]: As above, `bare-mental` -> `bare-metal`.

Also, mentioned tutorial covers only oneCCL, we need for example MKL as well. I personally prefer instructions on the [main IPEX page](https://intel.github.io/intel-extension-for-pytorch/index.html#installation?platform=gpu&version=v2.1.30%2bxpu&os=linux%2fwsl2&package=pip).

File: benchmark/multi_gpu/training/README.md [hunk 3]
```diff
@@ -10,15 +10,42 @@ python training_benchmark_cuda.py --dataset ogbn-products --model edge_cnn --num
...
+
+- Install [Intel® oneCCL Bindings for PyTorch](https://github.com/intel/torch-ccl) and [Intel® oneAPI Collective Communications Library (oneCCL)](https://www.intel.com/content/www/us/en/developer/tools/oneapi/oneccl.html)
+
+```bash
+# Install oneCCL package
+sudo apt install intel-oneapi-ccl-devel=2021.12.0-309
```
Comment (DamianSzwichtenberg) [inline benchmark/multi_gpu/training/README.md:32]: Add info about MKL/DPCPP

File: benchmark/multi_gpu/training/README.md [hunk 4]
```diff
@@ -10,15 +10,42 @@ python training_benchmark_cuda.py --dataset ogbn-products --model edge_cnn --num
...
+pip install torch==2.1.0.post2 intel-extension-for-pytorch==2.1.30+xpu --extra-index-url https://pytorch-extension.intel.com/release-whl/stable/xpu/us/
 ```
 
+### Running benchmark
+
 Run benchmark, e.g. assuming you have `n` XPUs:
 
 ```
-mpirun -np <n> python training_benchmark_xpu.py --dataset ogbn-products --model edge_cnn --num-epochs 3
+mpirun -n <n> python training_benchmark_xpu.py --dataset ogbn-products --model edge_cnn --num-epochs 3
```
Comment (DamianSzwichtenberg) [inline benchmark/multi_gpu/training/README.md:50]: `-np` appears in all official docs, let's keep it.

File: benchmark/multi_gpu/training/README.md (hunk 1 above)
Comment (zhouyu5) [reply benchmark/multi_gpu/training/README.md:24]: fixed.

File: benchmark/multi_gpu/training/README.md (hunk 3 above)
Comment (zhouyu5) [reply benchmark/multi_gpu/training/README.md:32]: Added.

File: benchmark/multi_gpu/training/README.md (hunk 2 above)
Comment (zhouyu5) [reply benchmark/multi_gpu/training/README.md:26]: Fixed and Added the link.

File: benchmark/multi_gpu/training/README.md (hunk 4 above)
Comment (zhouyu5) [reply benchmark/multi_gpu/training/README.md]: Ok. Revert it. BTW, `-np` and `-n` are equivalent. See belows:
image

**Comment by zhouyu5** [comment]
@DamianSzwichtenberg Thanks for reviewing. Please check the response inline.

**Comment by DamianSzwichtenberg** [comment]
LGTM.

**Review by DamianSzwichtenberg — APPROVED** [review]

**Pull request merged**